proxy/src/main/java/com/velocitypowered/proxy/config/VelocityConfiguration.java

```diff
@@ -151,22 +151,41 @@ public boolean validate() {
           + "receive any support!");
     }
 
-    switch (playerInfoForwardingMode) {
+    boolean requireForwardingSecret = false;
+    for (Map.Entry<String, PlayerInfoForwarding> entry : servers.getServerForwardingModes().entrySet()) {
+      switch (entry.getValue()) {
+        case NONE:
+          logger.warn("Player info forwarding is disabled for " + entry.getKey() + "!"
+              + " All players will appear to be connecting from the proxy and will have offline-mode UUIDs.");
+          break;
+        case MODERN:
+        case BUNGEEGUARD:
+          requireForwardingSecret = true;
+          break;
+        default:
+          break;
+      }
+    }
+
+    switch (defaultPlayerInfoForwardingMode) {
       case NONE:
-        logger.warn("Player info forwarding is disabled! All players will appear to be connecting "
+        logger.warn("Player info forwarding is disabled by default! All players will appear to be connecting "
             + "from the proxy and will have offline-mode UUIDs.");
         break;
       case MODERN:
       case BUNGEEGUARD:
-        if (forwardingSecret == null || forwardingSecret.length == 0) {
-          logger.error("You don't have a forwarding secret set. This is required for security.");
-          valid = false;
-        }
+        requireForwardingSecret = true;
         break;
       default:
         break;
     }
 
+    if (requireForwardingSecret && (forwardingSecret == null || forwardingSecret.length == 0)) {
+      logger.error("You don't have a forwarding secret set. This is required for security. "
+          + "See https://docs.papermc.io/velocity/player-information-forwarding for more details.");
```

Comment:
```suggestion
          + "See https://docs.papermc.io/velocity/player-information-forwarding#configuring-modern-forwarding for more details.");
```

Though this may be too long.